File: server/Models/Movie.cs
using System.ComponentModel.DataAnnotations;

namespace MovieApp.Server.Models;

public class Movie
{
    public int Id { get; set; }
    
    [Required]
    [MaxLength(200)]
    public string Title { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(100)]
    public string Director { get; set; } = string.Empty;
    
    [Required]
    [MaxLength(50)]
    public string Genre { get; set; } = string.Empty;
    
    [Required]
    public DateOnly ReleaseDate { get; set; }
    
    public int? Duration { get; set; } // Duration in minutes
    
    [Range(0.0, 10.0)]
    public decimal? Rating { get; set; }
    
    public string? Description { get; set; }
    
    [Url]
    public string? PosterUrl { get; set; }
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: app-ui/src/types/movie.ts
export interface Movie {
  id: number;
  title: string;
  director: string;
  genre: string;
  releaseDate: string; // DateOnly serialized as string
  duration?: number;
  rating?: number;
  description?: string;
  posterUrl?: string;
  createdAt: string;
  updatedAt: string;
}

export interface CreateMovieData {
  title: string;
  director: string;
  genre: string;
  releaseDate: string;
  duration?: number;
  rating?: number;
  description?: string;
  posterUrl?: string;
}

export interface UpdateMovieData {
  title?: string;
  director?: string;
  genre?: string;
  releaseDate?: string;
  duration?: number;
  rating?: number;
  description?: string;
  posterUrl?: string;
}

export interface MoviesResponse {
  movies: Movie[];
  pagination: {
    page: number;
    pageSize: number;
    totalCount: number;
    totalPages: number;
    hasNextPage: boolean;
    hasPreviousPage: boolean;
  };
}

export interface MoviesQueryParams {
  search?: string;
  genre?: string;
  page?: number;
  pageSize?: number;
}

export interface ApiError {
  type: string;
  title: string;
  status: number;
  detail?: string;
}